lib/rucio/common/types/dicts/typed_dict.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+__all__ = ['TypedDict', 'is_typeddict']
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict, is_typeddict
+else:
+    from typing_extensions import TypedDict, is_typeddict
```

Comment:
The reason for this file is to serve as a single source for the `TypedDict` class. An alternative to this is to just unconditionally import from `typing_extensions` everywhere we use it.

Reply:
Pyright seem to be confused when `TypedDict` is imported from `.typed_dict` instead of from `typing_extensions`, so I will remove this forwarding and use `typing_extensions` everywhere.